src/align.rs

```diff
@@ -186,15 +186,22 @@ impl<T: Copy, C: AlignedByteChunk> AlignedVec<T, C> {
         let old_len = self.len();
 
         // Resize the underlying vector to have enough chunks for the new length.
-        //
-        // NOTE: We don't need to `drop` any elements if the `Vec` is truncated since `T: Copy`.
-        let new_bytes = mem::size_of::<T>() * new_len;
+        // SAFETY: The `new_bytes` calculation must not overflow, ensuring a mathematical match
+        // with the underlying `inner` buffer size. NOTE: one can still pass ludicrous requested
+        // buffer lengths, just not unsound ones.
+        let Some(new_bytes) = mem::size_of::<T>().checked_mul(new_len) else {
+            panic!("Resizing would overflow the underlying aligned buffer");
+        };
```

Comment:
```suggestion
        let new_bytes = mem::size_of::<T>().checked_mul(new_len)
            .expect("Resizing would overflow the underlying aligned buffer");
```

If we aren't gonna do any fancy `#[inline(never)]`/`#[cold]` inner `fn` stuff for the panic, I'd rather just use `.expect` here.

(Also not sure if the formatting is right).

Reply:
Feel free to push that change to the fork.